ChefExtensionHandler/bin/chef-install.sh

```diff
@@ -145,7 +106,74 @@ install_from_repo(){
   yum -y install chef
   echo "Package Installed successfully ..."
 }
- 
+
+install_from_repo_ubuntu() {
+	platform="ubuntu"
+	curl_check $platform
+
+	# Need to first run apt-get update so that apt-transport-https can be installed
+	echo -n "Running apt-get update... "
+	apt-get update
+	echo "done."
+
+	echo -n "Installing apt-transport-https... "
+	apt-get install -y apt-transport-https
+	echo "done."
+
+	apt_config_url="https://packagecloud.io/install/repositories/chef/stable/config_file.list?os=ubuntu&dist=trusty&source=script"
+	apt_source_path="/etc/apt/sources.list.d/chef_stable.list"
+
+	echo -n "Installing $apt_source_path..."
+
+	# create an apt config file for this repository
+	curl -sSf "${apt_config_url}" > $apt_source_path
+	curl_exit_code=$?
+
+	if [ "$curl_exit_code" = "22" ]; then
+  	echo -n "Unable to download repo config from: "
+	  echo "${apt_config_url}"
+	  echo
+	  echo "Please contact support@packagecloud.io and report this."
+	  [ -e $apt_source_path ] && rm $apt_source_path
+  	exit 1
+	elif [ "$curl_exit_code" = "35" ]; then
+  	echo "curl is unable to connect to packagecloud.io over TLS when running: "
+	  echo "    curl ${apt_config_url}"
+	  echo "This is usually due to one of two things:"
+	  echo
+	  echo " 1.) Missing CA root certificates (make sure the ca-certificates package is installed)"
+  	echo " 2.) An old version of libssl. Try upgrading libssl on your system to a more recent version"
+	  echo
+  	echo "Contact support@packagecloud.io with information about your system for help."
+	  [ -e $apt_source_path ] && rm $apt_source_path
+  	exit 1
+	elif [ "$curl_exit_code" -gt "0" ]; then
+  	echo
+	  echo "Unable to run: "
+  	echo "    curl ${apt_config_url}"
+	  echo
+  	echo "Double check your curl installation and try again."
+	  [ -e $apt_source_path ] && rm $apt_source_path
+  	exit 1
+	else
+  	echo "done."
+	fi
+
+	echo -n "Importing packagecloud gpg key... "
+	# import the gpg key
+	curl https://packagecloud.io/gpg.key | sudo apt-key add -
+	echo "done."
+
+	echo -n "Running apt-get update... "
+	# update apt on this system
+	apt-get update
+	echo "done."
+
+  echo "Installing chef-client package"
+	apt-get install chef
```

Comment:
@NimishaS  we can refactor this code since we are using same logic for centos to the curl command status